src/main/java/core/basesyntax/strategy/ReportStrategy.java

```diff
@@ -0,0 +1,8 @@
+package core.basesyntax.strategy;
+
+import core.basesyntax.model.FruitTransaction;
+
+@FunctionalInterface
+public interface ReportStrategy {
+    void process(FruitTransaction transaction);
```

Comment:
Strategy is responsible only for returning the corresponding handler, handling of the transaction should be done outside of the strategy
```suggestion
public interface ReportStrategy {
    OperationHandler getHandlerByOperation(Operation operation);
```